src/Controller/BackendCsvImportController.php

```diff
@@ -0,0 +1,360 @@
+
+    /**
+     * @var string
+     */
+    private $contaoRoot;
+
+    /**
+     * Constructor.
+     *
+     * @param ContaoFrameworkInterface $framework
+     * @param Connection               $connection
+     * @param RequestStack             $requestStack
+     * @param string                   $contaoRoot
+     */
+    public function __construct(ContaoFrameworkInterface $framework, Connection $connection, RequestStack $requestStack, $contaoRoot)
+    {
+        $this->framework = $framework;
+        $this->connection = $connection;
+        $this->requestStack = $requestStack;
+        $this->contaoRoot = $contaoRoot;
+    }
+
+    /**
+     * Imports CSV data in the list wizard.
+     *
+     * @param DataContainer $dc
+     *
+     * @return Response
+     */
+    public function importListWizard(DataContainer $dc)
+    {
+        return $this->importFromTemplate(
+            function ($data, $row) {
+                return array_merge($data, $row);
+            },
+            $dc->table,
+            'listitems',
+            $dc->id,
+            $GLOBALS['TL_LANG']['MSC']['lw_import'][0],
+            true
+        );
+    }
+
+    /**
+     * Imports CSV data in the table wizard.
+     *
+     * @param DataContainer $dc
+     *
+     * @return Response
+     */
+    public function importTableWizard(DataContainer $dc)
+    {
+        return $this->importFromTemplate(
+            function ($data, $row) {
+                $data[] = $row;
+
+                return $data;
+            },
+            $dc->table,
+            'tableitems',
+            $dc->id,
+            $GLOBALS['TL_LANG']['MSC']['tw_import'][0]
+        );
+    }
+
+    /**
+     * Imports CSV data in the options wizard.
+     *
+     * @param DataContainer $dc
+     *
+     * @return Response
+     */
+    public function importOptionWizard(DataContainer $dc)
+    {
+        return $this->importFromTemplate(
+            function ($data, $row) {
+                $data[] = [
+                    'value' => $row[0],
+                    'label' => $row[1],
+                    'default' => $row[2] ? 1 : '',
+                    'group' => $row[3] ? 1 : '',
+                ];
+
+                return $data;
+            },
+            $dc->table,
+            'options',
+            $dc->id,
+            $GLOBALS['TL_LANG']['MSC']['ow_import'][0]
+        );
+    }
+
+    /**
+     * Runs the default import routine with a Contao template.
+     *
+     * @param callable    $callback
+     * @param string      $table
+     * @param string      $field
+     * @param int         $id
+     * @param string|null $submitLabel
+     * @param bool        $allowLinebreak
+     *
+     * @return Response
+     */
+    protected function importFromTemplate(callable $callback, $table, $field, $id, $submitLabel = null, $allowLinebreak = false)
+    {
+        $this->framework->initialize();
+
+        $request = $this->requestStack->getCurrentRequest();
```

Comment:
throw an exception? We can't handle that case, it cannot happen in the Contao backend.